codeforboston/vaccinatema
# Add support for i18n!

This PR adds `next-i18next`, "a plugin for Next.js projects that allows you to get translations up and running quickly and easily." This library is specific to Next.js (which we're already using), is actively maintained, has solid adoption, and was quick and easy to set up. Our i18n needs are pretty basic, but we can always swap out libraries if this one isn't sufficient.

This PR also adds the string "Eligibility" as a translatable string.

Not in this PR, but as a proof of concept, I added `locales/fr/common.json` and changed `dev.js` to:
```
import Layout from "../components/Layout"
import { i18n, withTranslation } from '../i18n'

const Home = ({t}) => (
  <Layout pageTitle="Home">
    <div>{t('welcome')}</div>
    <button type='button' onClick={() => i18n.changeLanguage(i18n.language === 'en' ? 'fr' : 'en')}></button>
  </Layout>
);

export default withTranslation('common')(Home);
```
![proofofconcept](https://user-images.githubusercontent.com/8495791/107975797-a6393580-6f86-11eb-83d1-461f60aae0c8.gif)

### Test Plan:
* Ensure main site and dev site function as usual.
* Ensure /dev/eligibility shows "Eligibility" as usual.
